controllers/object_controls.go

```diff
@@ -1213,6 +1265,36 @@ func TransformToolkit(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, n
 			}
 		}
 	}
+
+	// Update the driver-root volume if a custom driver install directory is configured with the operator
+	if config.HostPaths.DriverInstallDir != "" && config.HostPaths.DriverInstallDir != DefaultDriverInstallDir {
+		for index, volume := range obj.Spec.Template.Spec.Volumes {
+			if volume.Name == "driver-root" {
+				obj.Spec.Template.Spec.Volumes[index].HostPath.Path = config.HostPaths.DriverInstallDir
+			}
+		}
+	}
+
+	// If a custom devPath is configured which is not equal to 'hostRoot/dev' or 'driverInstallDir/dev', make it available
+	// to the toolkit container under '/dev-root/dev'
+	devRoot := filepath.Dir(config.HostPaths.DevPath)
+	if config.HostPaths.DevPath != "" && devRoot != config.HostPaths.DriverInstallDir && devRoot != config.HostPaths.Root {
+		devVolume := corev1.Volume{
+			Name: "custom-dev",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: filepath.Join(devRoot, "/dev"),
+				},
+			},
+		}
+		obj.Spec.Template.Spec.Volumes = append(obj.Spec.Template.Spec.Volumes, devVolume)
+
+		devVolumeMount := corev1.VolumeMount{
+			Name:      "custom-dev",
+			MountPath: "/dev-root/dev",
+		}
+		obj.Spec.Template.Spec.Containers[0].VolumeMounts = append(obj.Spec.Template.Spec.Containers[0].VolumeMounts, devVolumeMount)
+	}
```

Comment:
@elezar thoughts on removing `devPath` from this PR? In our driver validator, we would look for a `/dev` filesystem at `driverInstallDir/dev` and at `hostRoot/dev` before performing the driver validation. This is similar to logic in our k8s-dra-driver. At the very least, removing `devPath` would simplify the changes in this PR and make it easier to review. If required, we can add `devPath` in a follow-up.

Reply:
@tariq1890 while at the moment it may be okay to remove this option, once our driver container supports customizing the "driver root" and device node path, it may be useful to explicitly set both of these paths. For example, instead of installing everything at `/run/nvidia/driver`, we can instruct the driver container to install the driver at `/opt/nvidia/driver` and device nodes at `/dev`, for example.

Reply:
Based on the initial set of use cases we are targeting, I believe removing `devPath` from the API is sufficient for the time being. 

The logic in the driver validator would remain the same. The only difference would be that before creating the `driver-ready` status file, we would determine the value for `devRoot`. If `driverInstallDir/dev` exists, then we set `devRoot = driverInstallDir`. Else we fallback to setting `devRoot = hostRoot`.

Below are the various use cases:

## Driver Container
Input:
```
driver:
  enabled: true
  
hostPaths:
  root: /
  driverInstallDir: /run/nvidia/driver
```
Resolved values:
```
NVIDIA_DRIVER_ROOT: /run/nvidia/driver
DRIVER_ROOT_CTR_PATH: /driver-root
NVIDIA_DEV_ROOT: /run/nvidia/driver
DEV_ROOT_CTR_PATH: /driver-root
```

## Driver Container with host driver
Input:
```
driver:
  enabled: true
  
hostPaths:
  root: /
  driverInstallDir: /run/nvidia/driver
```
Resolved values:
```
NVIDIA_DRIVER_ROOT: /
DRIVER_ROOT_CTR_PATH: /host
NVIDIA_DEV_ROOT: /
DEV_ROOT_CTR_PATH: /host
```

## GKE driver
Input:
```
driver:
  enabled: false
  
hostPaths:
  root: /
  driverInstallDir: /home/kubernetes/bin/nvidia
```
Resolved values:
```
NVIDIA_DRIVER_ROOT: /home/kubernetes/bin/nvidia
DRIVER_ROOT_CTR_PATH: /driver-root
NVIDIA_DEV_ROOT: /
DEV_ROOT_CTR_PATH: /host
```

Reply:
I have removed `devPath` from the ClusterPolicy API now.